### ISSUE TITLE
Add plugin event for before launching browser

### ISSUE THREAD

**Issue by chrisbreiding** [cypress-io/cypress#691]
Will allow updating flags sent to browser, with the following signature:

```js
register('on:before:browser:launch', (browserName, flags) => {
  return flags
})
```

The modified flags should be returned or resolved in a promise.

**Comment by jennifer-shehane** [comment]
I had a use case that I wanted to mention here. We had a user that asked about testing `@media print` css stylings. There is a chromium flag - [`enable-print-browser`](https://peter.sh/experiments/chromium-command-line-switches/#enable-print-browser) that would have the browser render as if printed which would allow them to test this.

**Comment by verheyenkoen** [comment]
Would that mean that all tests are ran in print mode or can you set the flag on a per-test-basis?

**Comment by brian-mann** [comment]
This will be accomplished by the #684 proposal.

Flags will only ever affect the browser once at launch. You'd have to separate out your tests to only run in that specific mode from the other regular ones.

**Comment by brian-mann** [comment]
Released in [`1.3.0`](https://docs.cypress.io/guides/references/changelog.html#1-3-0).
